vfs/read_write.go

```diff
@@ -260,7 +261,21 @@ func (fh *RWFileHandle) _readAt(b []byte, off int64, release bool) (n int, err e
 		// Do the writing with fh.mu unlocked
 		fh.mu.Unlock()
 	}
-	n, err = fh.item.ReadAt(b, off)
+	for {
+		fh.item.ProtectCache()
```

Comment:
I'm not a big fan of the `ProtectCache` calls - they are an implementation detail for the cache, so can we push them into the item.ReadAt call and any others which need it?

Reply:
Agreed.  I also noticed that and will move them into item.ReadAt.